wpimath/src/main/java/edu/wpi/first/math/filter/SlewRateLimiter.java

```diff
@@ -14,53 +14,107 @@
-   */
-  public SlewRateLimiter(double rateLimit, double initialValue) {
-    m_rateLimit = rateLimit;
-    m_prevVal = initialValue;
-    m_prevTime = WPIUtilJNI.now() * 1e-6;
-  }
+    /**
+     * Creates a new SlewRateLimiter with the given rate limit and initial value.
+     *
+     * @param rateLimit    The rate-of-change limit, in units per second.
+     * @param initialValue The initial value of the input.
+     */
+    public SlewRateLimiter(double rateLimit, double initialValue) {
+        m_rateLimit = rateLimit;
+        m_prevVal = initialValue;
+        m_prevTime = WPIUtilJNI.now() * 1e-6;
+    }
 
-  /**
-   * Creates a new SlewRateLimiter with the given rate limit and an initial value of zero.
-   *
-   * @param rateLimit The rate-of-change limit, in units per second.
-   */
-  public SlewRateLimiter(double rateLimit) {
-    this(rateLimit, 0);
-  }
+    /**
+     * Creates a new SlewRateLimiter with the given rate limit and an initial value of zero.
+     *
+     * @param rateLimit The rate-of-change limit, in units per second.
+     */
+    public SlewRateLimiter(double rateLimit) {
+        this(rateLimit, 0);
+    }
 
-  /**
-   * Filters the input to limit its slew rate.
-   *
-   * @param input The input value whose slew rate is to be limited.
-   * @return The filtered value, which will not change faster than the slew rate.
-   */
-  public double calculate(double input) {
-    double currentTime = WPIUtilJNI.now() * 1e-6;
-    double elapsedTime = currentTime - m_prevTime;
-    m_prevVal +=
-        MathUtil.clamp(input - m_prevVal, -m_rateLimit * elapsedTime, m_rateLimit * elapsedTime);
-    m_prevTime = currentTime;
-    return m_prevVal;
-  }
+    /**
+     * Creates a new SlewRateLimiter with the given forward rate and backward rate.
+     *
+     * @param forward_Rate
+     * @param backwardRateLimit
+     */
+    public SlewRateLimiter(double forwardRateLimit, double backwardRateLimit) {
+        m_forwardRateLimit = forwardRateLimit;
+        m_backwardRateLimit = backwardRateLimit;
+        m_prevTime = WPIUtilJNI.now() * 1e-6;
+    }
 
-  /**
-   * Resets the slew rate limiter to the specified value; ignores the rate limit when doing so.
-   *
-   * @param value The value to reset to.
-   */
-  public void reset(double value) {
-    m_prevVal = value;
-    m_prevTime = WPIUtilJNI.now() * 1e-6;
-  }
 }
+
+    /**
+     * Filters the input to limit its slew rate.
+     *
+     * @param input The input value whose slew rate is to be limited.
+     * @return The filtered value, which will not change faster than the slew rate.
+     */
+    public double calculate(double input) {
+        double currentTime = WPIUtilJNI.now() * 1e-6;
+        double elapsedTime = currentTime - m_prevTime;
+        m_prevVal +=
+                MathUtil.clamp(input - m_prevVal, -m_rateLimit * elapsedTime, m_rateLimit * elapsedTime);
+        m_prevTime = currentTime;
+        return m_prevVal;
+    }
+
```

Comment:
```suggestion
    /**
     * Filters the input to limit its slew rate.
     *
     * @param input The input value whose slew rate is to be limited.
     * @return The filtered value, which will not change faster than the slew rate.
     */
    public double calculate(double input) {
        double currentTime = WPIUtilJNI.now() * 1e-6;
        double elapsedTime = currentTime - m_prevTime;
        m_prevVal +=
                MathUtil.clamp(input - m_prevVal, -m_rateLimit * elapsedTime, m_rateLimit * elapsedTime);
        m_prevTime = currentTime;
        return m_prevVal;
    }
```

Looks like this got copied again? This is the old one; it should be removed (this code won't even compile at the moment).